src/components/monitor/list-monitor-scroller.jsx

```diff
@@ -0,0 +1,109 @@
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import bindAll from 'lodash.bindall';
+
+import styles from './monitor.css';
+import {List} from 'react-virtualized';
+
+class ListMonitorScroller extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'rowRenderer',
+            'noRowsRenderer',
+            'handleEventFactory'
+        ]);
+    }
+    handleEventFactory (index) {
+        return () => this.props.onActivate(index);
+    }
+    noRowsRenderer () {
+        return (
+            <div className={styles.listEmpty}>
+                {'(empty)' /* TODO waiting for design before translation */}
+            </div>
+        );
+    }
+    rowRenderer ({index, key, style}) {
+        return (
+            <div
+                className={styles.listRow}
+                key={key}
+                style={style}
+            >
+                <div className={styles.listIndex}>{index + 1 /* one indexed */}</div>
+                <div
+                    className={styles.listValue}
+                    dataIndex={index}
+                    style={{background: this.props.categoryColor}}
+                    onClick={this.handleEventFactory(index)}
+                >
+                    {this.props.activeIndex === index ? (
+                        <div className={styles.inputWrapper}>
+                            <input
+                                autoFocus
+                                autoComplete={false}
+                                className={classNames(styles.listInput, 'no-drag')}
+                                spellCheck={false}
+                                type="text"
+                                value={this.props.activeValue} /* eslint-disable-line */
+                                onBlur={this.props.onDeactivate}
+                                onChange={this.props.onInput}
+                                onFocus={this.props.onFocus}
+                                onKeyDown={this.props.onKeyPress} // key down to get ahead of blur
+                            />
+                            <div
+                                className={styles.removeButton}
+                                onMouseDown={this.props.onRemove} // mousedown to get ahead of blur
+                            >
+                                {'✖︎'}
+                            </div>
+                        </div>
+
+                    ) : (
+                        <div className={styles.valueInner}>{this.props.values[index]}</div>
+                    )}
+                </div>
+            </div>
+        );
+    }
+    render () {
+        const {height, values, width, activeIndex, activeValue} = this.props;
+        // Keep the active index in view if defined, else must be undefined for List component
+        const scrollToIndex = activeIndex === null ? undefined : activeIndex; /* eslint-disable-line no-undefined */
```

Comment:
why set to undefined and not null?

Reply:
Sorry I tried to make it clear in the comment above. The component I'm using (List) takes scrollToIndex as an optional prop (which means it'll be undefined if it isn't passed in). Apparently they are looking specifically for undefineds, and `null` is not an acceptable value.